### PR TITLE
DOC: More single to double backtick

### PR DESCRIPTION
**Pull request by Carreau**
A quick scan of sphinx errors when building with default role being `py:obj`, a few references look impossible as they are invalid python identifiers, and most of them need double backticks.

### CODE REVIEW

File: scipy/optimize/_lbfgsb_py.py [hunk 1]
```diff
@@ -201,8 +201,8 @@ def fmin_l_bfgs_b(func, x0, fprime=None, args=(),
     array([1.99999999, 3.00000006]), 1.7746231151323805e-14  # may vary
 
     The optimized parameters in ``x_opt`` agree with the ground truth parameters
-    ``m`` and ``b``. Next, let us perform a bound contrained optimization using the `bounds`
-    parameter. 
+    ``m`` and ``b``. Next, let us perform a bound constrained optimization using
+    the `bounds` parameter.
```
Comment (Carreau) [inline scipy/optimize/_lbfgsb_py.py:205]: rewrapped as otherwise the exrtra backticks were making the lines too long.

**Review by j-bowhay — APPROVED** [review]
Thanks @Carreau

**Comment by j-bowhay** [comment]
ps. for future doc only change feel free to add `[docs only]` to the end of the commit message to save on CI

**Comment by Carreau** [comment]
>  ps. for future doc only change feel free to add [docs only] to the end of the commit message to save on CI

Thanks for merging, and yes i'm supposed to know that, but I keep forgetting...

**Pull request merged**